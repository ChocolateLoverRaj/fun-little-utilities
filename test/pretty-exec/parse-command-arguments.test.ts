import {
  ParsingResult,
  parseSingleCommandArgument,
  parseCommandArguments,
} from 'pretty-exec'

describe('parseSingleCommandArguments', () => {
  it('"argument"', () => {
    expect(parseSingleCommandArgument('argument')).toEqual({
      type: ParsingResult.Type.Argument,
      text: 'argument',
    })
  })

  it('"--flag"', () => {
    expect(parseSingleCommandArgument('--flag')).toEqual({
      type: ParsingResult.Type.Flag,
      text: '--flag',
    })
  })

  it('"-flag"', () => {
    expect(parseSingleCommandArgument('-flag')).toEqual({
      type: ParsingResult.Type.Flag,
      text: '-flag',
    })
  })

  it('"--option-key=option-value"', () => {
    expect(parseSingleCommandArgument('--option-key=option-value')).toEqual({
      type: ParsingResult.Type.Option,
      text: '--option-key=option-value',
      key: '--option-key',
      value: 'option-value',
    })
  })
})

describe('parseCommandArguments', () => {
  it('matches snapshot', () => {
    const args = ['-o', 'output', '--force', '--foo=bar', '--', 'abc', '--not-flag']
    expect(parseCommandArguments(args)).toMatchSnapshot()
  })
})
