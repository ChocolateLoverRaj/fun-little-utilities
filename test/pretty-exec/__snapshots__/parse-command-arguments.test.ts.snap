// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseCommandArguments matches snapshot 1`] = `
Array [
  Object {
    "text": "-o",
    "type": "Flag",
  },
  Object {
    "text": "output",
    "type": "Argument",
  },
  Object {
    "text": "--force",
    "type": "Flag",
  },
  Object {
    "key": "--foo",
    "text": "--foo=bar",
    "type": "Option",
    "value": "bar",
  },
  Object {
    "text": "--",
    "type": "Flag",
  },
  Object {
    "text": "abc",
    "type": "Argument",
  },
  Object {
    "text": "--not-flag",
    "type": "Argument",
  },
]
`;
